include/metall/detail/file.hpp

```diff
@@ -519,11 +409,11 @@ inline bool get_regular_file_names(const std::string &dir_path,
 /// If <= 0 is given, the value is automatically determined.
 /// \param copy_func The actual copy function.
 /// \return  On success, returns true. On error, returns false.
+template<typename F>
 inline bool copy_files_in_directory_in_parallel_helper(
```

Comment:
If something goes wrong the output folder is a mess. To me, it would make sense to (1) report what files were successfully copied or (2) roll back all copies if one failed.

Reply:
The user can tell if the copy failed because the top level function returns false. 
What they do with that information is up to them I would say. Doesn't really give us any benefit to decide for them what to do.

Reply:
I think some of the usual reactions could be:
- ignore that it failed but report
- repeat however many times you want and then abort or something

Reply:
typical reason for failing would be lack of storage. Repeating would not help. I would rather abort the program or operation. But yes, that's up to the application. Here, it's mainly used for snapshots if I remember correctly. So, we probably don't need to to more here.